core/translate.c

```diff
@@ -330,12 +334,16 @@ translate_walk_track(dcontext_t *tdcontext, instr_t *inst, translate_walk_t *wal
                     spill = false;
             });
             /* if a restore whose spill was before a cti, ignore */
-            if (spill || walk->reg_spilled[r]) {
-                /* ensure restores and spills are properly paired up */
-                ASSERT((spill && !walk->reg_spilled[r]) ||
-                       (!spill && walk->reg_spilled[r]));
+            if (spill || walk->reg_spilled[r] != UINT_MAX) {
+                /* Ensure restores and spills are properly paired up, but we do
+                 * allow for redundant spills.
+                 */
+                ASSERT(spill || (!spill && walk->reg_spilled[r] != UINT_MAX));
                 ASSERT(spill || walk->reg_tls[r] == spill_tls);
-                walk->reg_spilled[r] = spill;
+                if (spill)
+                    walk->reg_spilled[r] = offs;
```

Comment:
I see here that this holds an offset: how about renaming it to reg_spill_offs?  IMHO that will make all the new code using it clearer and easier to read.

Reply:
Done.